
<launch>

  # 이동로봇 상태 관리
  <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find shared_control)/urdf/turtlebot3_waffle_gamma.urdf.xacro'"/>
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="state_publisher"/>
  <node pkg="turtlebot3_bringup" type="turtlebot3_diagnostics" name="turtlebot3_diagnostics"/>

  # OpenCR 연결
  <node pkg="rosserial_python" type="serial_node.py" name="turtlebot3_core">
    <param name="port" value="/dev/ttyACM0"/>
    <param name="baud" value="115200"/>
    <remap from="imu" to="imu_raw"/>
  </node>
  <node pkg="shared_control" type="filter.py" name="filter">
    <param name="imu_in" value="imu_raw"/>
    <param name="imu_out" value="imu"/>
    <param name="imu_delayed" value="true"/>
  </node>

  # RPLidar 연결
  <node name="rplidarNode" pkg="rplidar_ros"  type="rplidarNode" output="screen">
    <param name="serial_port"      type="string" value="/dev/ttyUSB0"/>
    <param name="serial_baudrate"  type="int"    value="256000"/>
    <param name="frame_id"         type="string" value="base_scan"/>
    <param name="inverted"         type="bool"   value="false"/>
    <param name="angle_compensate" type="bool"   value="true"/>
    <param name="scan_mode"        type="string" value="Sensitivity"/>
  </node>

  # Realsense 연결
  <include file="$(find shared_control)/launch/realsense.launch"/>

</launch>
